api-v2/src/student/student.module.ts

```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
+import { StudentService } from './student.service';
+import { StudentController } from './student.controller';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Student } from './entities/student.entity';
+
+@Module({
+  imports: [TypeOrmModule.forFeature([Student])],
```

Comment:
rather than having to put this everywhere, could alternatively do it once in app.module.ts if you don't mind not specifying specific feature, [see here](https://github.com/sandboxnu/office-hours/blob/cca52eb54092abe674fb53f3086453bc61613b94/packages/server/src/app.module.ts#L26)

Reply:
From the file you linked i think we have it setup similarly as well in app.module(correct me i'm wrong though!). The `connection` object is available throughout the application without requiring feature imports but I find the repository pattern much better because 
- it explicitly requires us to think about what tables each module interacts with 
- kinda follows the least privilege access principle. using the entire typeorm connection object is way too much functionality that most providers don't need. we mostly simply need the repository(s) for the table(s) we are accessing/updating
- i tend to like the repository pattern in general as a good way of using typeorm in any project since it abstracts away unnecessary db terms like `connection` and gives us a nice/simple interface that lets us do things with a specific table
Do the points make sense?